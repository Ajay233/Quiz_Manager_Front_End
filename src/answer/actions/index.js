import { getUsingParams, post, put, del } from '../../axiosRequests/requests'
import { setNotification } from '../../notifications/actions'
import { hideModal } from '../../modal/actions'
import { sessionExpired } from '../../utils/session'
import history from '../../history'

// create
export const addAnswer = (data, jwt) => {
  return (dispatch) => {
    return post("answer/create", [data], jwt).then((response) => {
      dispatch({
        type: "ADD_ANSWER",
        payload: response.data[0]
      })
      history.push("/editQuestion");
      dispatch(setNotification("Answer created", "success", true));
    }).catch((error) => {
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        console.log(error.response);
        dispatch(setNotification("Error - unable to create answer", "error", true));
      }
    });
  }
}

// read
export const getAnswers = (param, jwt) => {
  return (dispatch) => {
    return getUsingParams("answer/findByQuestionId", param, jwt).then((response) => {
      dispatch({
        type: "SET_ANSWERS",
        payload: response.data
      });
    }).catch((error) => {
      dispatch(setNotification("Error retrieving answers", "error", true));
    });
  }
}

// update (there was a call to setCurrentAnswer but have removed this as it's not needed)
export const updateAnswer = (body, jwt) => {
  return (dispatch) => {
    return put("answer/update", [body], jwt).then((response) => {
      dispatch({
        type: "UPDATE_ANSWER",
        payload: response.data[0]
      })
      history.push("/editQuestion");
      dispatch(setNotification("Answer updated", "success", true));
    }).catch((error) => {
      console.log(error.response);
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        dispatch(setNotification("Error - unable to update answer", "error", true));
      }
    });
  }
}

// delete
export const deleteAnswer = (config, jwt) => {
  return (dispatch) => {
    return del("answer/delete", config, jwt).then((response) => {
      dispatch(hideModal())
      dispatch({
        type: "DELETE_ANSWER",
        payload: config.data[0]
      })
      dispatch(setNotification("Answer deleted", "success", true));
    }).catch((error) => {
      console.log(error.response);
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        dispatch(hideModal())
        dispatch(setNotification("Error - Unable to delete this answer", "error", true))
      }
    });
  }
}

export const setCurrentAnswer = (answer) => {
  return {
    type: "SET_CURRENT_ANSWER",
    payload: answer
  }
}
