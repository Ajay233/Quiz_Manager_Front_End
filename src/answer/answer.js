import React from 'react'
import { Link } from 'react-router-dom'

export class Answer extends React.Component {

  handleEdit = () => {
    this.props.setCurrentAnswer(this.props.answer);
  }

  handleDelete = () => {
    this.props.showModal();
    this.props.setCurrentAnswer(this.props.answer);
  }

  renderAnswer = () => {
    const { answerNumber, description, correctAnswer } = this.props.answer
    return(
      <div className="answerContainer">
        <div className="ansNumber">{ answerNumber }</div>
        <div className="ansDecription">{ description }</div>
        <div className="correctAns">{ correctAnswer === true ? <i className="far fa-check-circle green"></i> : <i className="far fa-times-circle red"></i> }</div>
        {this.renderOptions()}
      </div>
    );
  }

  renderOptions = () => {
    const { permission } = this.props.userData;
    return(
      <div className="options">
        { permission === "ADMIN" || permission === "SUPER-USER" ? <Link to="#" className="deleteOption" onClick={this.handleDelete}><i className="fas fa-trash-alt red"></i> Delete</Link> : null }
        { permission === "ADMIN" || permission === "SUPER-USER" ? <Link to="/editAnswer" className="edit" onClick={this.handleEdit}><i className="fas fa-edit blue"></i> Edit</Link> : null }
      </div>
    );
  }

  render(){
    return(
      <div>
        {this.renderAnswer()}
      </div>
    );
  }
}

export default Answer
