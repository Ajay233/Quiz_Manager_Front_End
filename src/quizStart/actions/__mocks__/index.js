export const addAnswer = jest.fn()

export const incrementQuestion = jest.fn()

export const showResults = jest.fn()

export const exitQuiz = jest.fn()

export const clearQuizProgress = jest.fn()
