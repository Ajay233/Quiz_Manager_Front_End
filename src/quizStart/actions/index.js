export const addAnswer = (answer) => {
  return {
    type: "ADD_SELECTED_ANSWER",
    payload: answer
  }
}

export const incrementQuestion = (questionNumber) => {
  return {
    type: "INCREMENT_QUESTION",
    payload: questionNumber
  }
}

export const showResults = () => {
  return {
    type: "SHOW_RESULTS",
    payload: true
  }
}

export const exitQuiz = () => {
  return {
    type: "RESET_STATE"
  }
}
