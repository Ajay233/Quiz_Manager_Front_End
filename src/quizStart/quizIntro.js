import React from 'react'
import { connect } from 'react-redux'
import { timedFunc } from '../utils/display'
import history from '../history'

class QuizIntro extends React.Component {

  componentDidMount(){
    document.documentElement.scrollTop = 0;
  }

  timedRedirect = () => {
    history.push("/startQuiz")
  }

  renderImg = (imgUrl) => {
    return imgUrl ? <div className="quizIntroImgContainer"><img src={imgUrl}  className="quizIntroImg" alt=""/></div> : null
  }

  renderAuthor = (author) => {
    return author !== 'Anonymous' ? <div className="quizIntroAuthor">{`Created by: ${author}`}</div> : null
  }

  render(){
    const { name, description, imgUrl, author } = this.props.quiz
    return(
      <div className="componentContainer">
        {timedFunc(2500, this.timedRedirect)}
        <div className="quizIntroTitle">{name}</div>
        {this.renderImg(imgUrl)}
        <div className="quizIntroDescription">{description}</div>
        {this.renderAuthor(author)}
      </div>
    );
  }
}

export const mapStateToProps = (state) => {
  return {
    quiz: state.quiz
  }
}

export default connect(mapStateToProps)(QuizIntro)
