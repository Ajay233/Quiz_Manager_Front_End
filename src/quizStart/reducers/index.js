import { quizProgressTracking } from '../factory/quizProgressTrackingFactory'

export const setQuizProgressTracking = (state=quizProgressTracking, action) => {
  switch (action.type) {
    case "ADD_SELECTED_ANSWER": return {...state, answersPicked: [...state.answersPicked, action.payload] };
    case "INCREMENT_QUESTION": return {...state, questionNumber: action.payload };
    case "SHOW_RESULTS": return {...state, showResults: action.payload};
    default: return state;
  }
}
