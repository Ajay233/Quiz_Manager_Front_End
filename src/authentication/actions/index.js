import { post } from '../../axiosRequests/requests'
import { setNotification } from '../../notifications/actions'
import history from '../../history'

export const setUser = (endpoint, loginDetails) => {

  return (dispatch) => {

    post(endpoint, loginDetails).then((response) => {
      dispatch({
        type: "SET_USER_LOGGED_IN",
        payload: {
          id: response.data.user.id,
          forename: response.data.user.forename,
          surname: response.data.user.surname,
          email: response.data.user.email,
          permission: response.data.user.permission,
          verified: response.data.user.verified,
          jwt: response.data.jwt,
          loggedIn: true,
          response: response
        }
      })
      history.push("/")
      dispatch(setNotification(`Welcome back ${response.data.user.forename}`, "loginSuccess", true))
    }).catch((error) => {
      if(error.response.data === "NOT VERIFIED"){
        const verifyMsg = "Your email has not yet been veirified.  You will need to verify your email before you can log in"
        dispatch(setNotification(verifyMsg, "warning", true))
      } else {
        dispatch({
          type: "SET_USER_LOGGED_IN",
          payload: {
            error: error.response
          }
        });
        dispatch(setNotification("The username or password you entered was incorrect", "error", true));
      }
    })
  }
}

export const logOutUser = () => {
  return {
    type: "LOG_OUT_USER",
    payload: {
      forename: "",
      surname: "",
      email: "",
      permission: "",
      verified: "",
      jwt: "",
      loggedIn: false
    }
  }
}

export const setVerficationProcess = (endpoint, verificationDetails) => {

  return (dispatch) => {

    post(endpoint, verificationDetails).then((response) => {
      dispatch({
        type: "SET_VERIFY_PROCESS_STATUS",
        payload: {
          completionStatus: "completed",
          token: "",
          error: {}
        }
      })
    }).catch((error) => {
      dispatch({
        type: "SET_VERIFY_PROCESS_STATUS",
        payload: {
          completionStatus: "not verified",
          token: verificationDetails.token,
          error: error.response
        }
      })
    })
  }
}

export const setRedirect = (status) => {
  return {
    type: "SET_REDIRECT",
    payload: {
      status: status
    }
  }
}

export const logOut = () => {
  return {
    type: "RESET_APP"
  }
}

// export const getUsers = (endpoint, header) => {
//   return {
//     type: "GET_USER",
//     payload: {
//       userId
//   }
// }
