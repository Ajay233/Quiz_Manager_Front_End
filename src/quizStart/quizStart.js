import React from 'react'
import { connect } from 'react-redux'
import PickAnswer from './forms/pickAnswer'
import { addAnswer, incrementQuestion, showResults, exitQuiz, clearQuizProgress } from './actions'
import { setCurrentQuestion } from '../question/actions'
import { getAnswers } from '../answer/actions'
import { setNotification } from '../notifications/actions'
import history from '../history'
import '../stylesheets/quizStart.css'

export class QuizStart extends React.Component {

  componentDidMount(){
    if(!this.props.userData.loggedIn){
      history.push('/login')
      this.props.setNotification("Your session has expired, please log in to continue", "warning", true)
    }
    window.scrollTo(500, 0);
  }

  componentWillUnmount(){
    this.props.clearQuizProgress()
  }

  handleSubmit = ({ answer }) => {
    const { questions } = this.props;
    const { questionNumber } = this.props.quizProgressTracking
    questionNumber + 1 < questions.length ? this.nextQuestion(answer) : this.addLastAnswer(answer);
  }

  nextQuestion = (answer) => {
    const { questions, addAnswer, answers, incrementQuestion, setCurrentQuestion, getAnswers, userData } = this.props;
    const { questionNumber } = this.props.quizProgressTracking
    const body = { questionId: questions[questionNumber + 1].id }
    const mappedAnswer = answers.filter(e => {return e.description === answer})
    addAnswer(mappedAnswer[0]);
    incrementQuestion(questionNumber + 1);
    setCurrentQuestion(questions[questionNumber + 1]);
    getAnswers(body, userData.jwt)
  }

  addLastAnswer = (answer) => {
    const { addAnswer, answers, incrementQuestion, showResults } = this.props;
    const { questionNumber } = this.props.quizProgressTracking
    const mappedAnswer = answers.filter(e => {return e.description === answer})
    addAnswer(mappedAnswer[0]);
    incrementQuestion(questionNumber + 1);
    showResults();
  }

  renderResults = () => {
    const { questions } = this.props;
    const { answersPicked, showResults } = this.props.quizProgressTracking
    const correctAnswers = answersPicked.filter(e => { return e.correctAnswer === true })
    return showResults ? <div id="quizResult" className="title-large"><b>{`You scored ${correctAnswers.length} out of ${questions.length}`}</b></div> : null;
  }

  renderQuestion = () => {
    const { currentQuestion } = this.props;
    return(
      <div id="questionContainer" className="sectionSpacing">
        <div className="title-medium-left-alt">{`Question ${currentQuestion.questionNumber}`}</div>
        <div>
          {currentQuestion.description}
        </div>
      </div>
    );
  }

  renderQuestionImg = () => {
    const { imgUrl } = this.props.currentQuestion
    if(imgUrl !== null){
      return(
        <div id="questionImageContainer" className="">
          <div className="imgWrapper">
            <img src={imgUrl} alt="" className="quizQuestionImg"/>
          </div>
        </div>
      );
    } else {
      return null
    }
  }

  render(){
    const { quiz, answers, questions, exitQuiz, currentQuestion } = this.props
    const { showResults, questionNumber } = this.props.quizProgressTracking
    return(
      <div className="componentContainer">
        {questions[1] === undefined ? null : console.log(questions[1].description)}
        <div id="quizStartTitle">{quiz.name}</div>
        <div id="quizStartContainer">
          <div className={currentQuestion.imgUrl === null ? "questionDetailsContainer" : "questionDetailsContainerAdjusted"}>
            {this.renderQuestion()}
            <img id="quizBot2" src={require("../public/icons/quizBot2.png")} alt="" />
            {this.renderResults()}
          </div>
          {this.renderQuestionImg()}
          <div className={currentQuestion.imgUrl === null ? "answerFormContainer" : "answerFormContainerAdjusted"}>
            <PickAnswer
              title={"Answers"}
              answers={answers}
              onSubmit={this.handleSubmit}
              numberOfQuestions={questions.length}
              currentQuestionNumber={questionNumber + 1}
              showResults={showResults}
              exit={exitQuiz}
            />
          </div>
        </div>
      </div>
    );
  }
}


export const mapStateToProps = (state) => {
  return{
    userData: state.userData,
    quiz: state.quiz,
    questions: state.questions,
    currentQuestion: state.currentQuestion,
    answers: state.answers,
    quizProgressTracking: state.quizProgressTracking
  }
}

export default connect(mapStateToProps,
  { addAnswer,
    setCurrentQuestion,
    getAnswers,
    incrementQuestion,
    showResults,
    exitQuiz,
    clearQuizProgress,
    setNotification
  })(QuizStart)
