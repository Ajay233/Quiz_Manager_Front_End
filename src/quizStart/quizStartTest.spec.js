import React from 'react'
import ReactDOM from 'react-dom'
import { QuizStart, mapStateToProps } from './quizStart'
import { Router } from 'react-router-dom'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import mockAxios from 'jest-mock-axios'
import { render, fireEvent, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import history from '../history'

import { addAnswer, incrementQuestion, showResults, exitQuiz, clearQuizProgress } from './actions'
import { setCurrentQuestion } from '../question/actions'
import { getAnswers } from '../answer/actions'
import { setNotification } from '../notifications/actions'

jest.mock("./actions")
jest.mock("../question/actions")
jest.mock("../answer/actions")
jest.mock("../notifications/actions")

const mockStore = configureStore({})

describe("mapStateToProps", () => {
  it("should map state to props", () => {
    const appState = {
      userData: { id: 1, loggedIn: true },
      quiz: { id: 1, name: "test", description: "test" },
      questions: [
        { id: 1, questionNumber: 1, description: "test" },
        { id: 2, questionNumber: 2, description: "test2" }
      ],
      currentQuestion: { id: 1, questionNumber: 1, description: "test" },
      answers: [
        { id: 1, answerIndex: 1, description: "test" },
        { id: 2, answerIndex: 2, description: "test2" }
      ],
      quizProgressTracking: { questionNumber: 0, answersPicked:[], showResults: false }
    }

    const componentState = mapStateToProps(appState)
    expect(componentState).toEqual(appState)
  })
})

describe("", () => {

  let store;
  let userData;
  let quiz;
  let questions;
  let currentQuestion;
  let answers;
  let quizProgressTracking;

  beforeEach(() => {
    store = mockStore({})

    userData = { id: 1, jwt: "jwt", loggedIn: true }
    quiz = { id: 1, name: "test", description: "test" }
    questions = [
      { id: 1, questionNumber: 1, description: "test" },
      { id: 2, questionNumber: 2, description: "test2" }
    ],
    currentQuestion = { id: 1, questionNumber: 1, description: "test" }
    answers = [
      { id: 1, answerIndex: 1, description: "test" },
      { id: 2, answerIndex: 2, description: "test2" }
    ],
    quizProgressTracking = { questionNumber: 0, answersPicked:[], showResults: false }

    window.scrollTo = jest.fn()
  })

  afterEach(() => {
    addAnswer.mockClear()
    incrementQuestion.mockClear()
  })

  it("should render a question, list of answers to pick, 'next question' and 'quit' buttons", () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <QuizStart
            userData={userData}
            quizProgressTracking={quizProgressTracking}
            questions={questions}
            quiz={quiz}
            currentQuestion={currentQuestion}
            answers={answers}
            setNotification={setNotification}
          />
        </Router>
      </Provider>
    );

    expect(component).toMatchSnapshot()
  })

  it("should render the results", () => {
    const finishedQuiz = {
      questionNumber: 2,
      answersPicked:[
        { id: 1, answerIndex: 1, description: "test", correctAnswer: true },
        { id: 2, answerIndex: 2, description: "test2", correctAnswer: false }
      ],
      showResults: true
    }

    const component = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <QuizStart
            userData={userData}
            quizProgressTracking={finishedQuiz}
            questions={questions}
            quiz={quiz}
            currentQuestion={currentQuestion}
            answers={answers}
            setNotification={setNotification}
          />
        </Router>
      </Provider>
    );

    expect(component).toMatchSnapshot()
  })

  describe("nextQuestion", () => {
    it("should call", () => {
      const pickedAns = { id: 1, answerIndex: 1, description: "test" }
      const body = { questionId: 2 }
      const component = render(
        <Provider store={store}>
          <Router history={history}>
            <QuizStart
              userData={userData}
              quizProgressTracking={quizProgressTracking}
              questions={questions}
              quiz={quiz}
              currentQuestion={currentQuestion}
              answers={answers}
              addAnswer={addAnswer}
              incrementQuestion={incrementQuestion}
              setCurrentQuestion={setCurrentQuestion}
              getAnswers={getAnswers}
              answer={pickedAns}
              clearQuizProgress={clearQuizProgress}
              setNotification={setNotification}
            />
          </Router>
        </Provider>
      );
      fireEvent.click(component.getByTestId("next-button"))
      expect(addAnswer).toHaveBeenCalledTimes(1)
      // expect(addAnswer).toHaveBeenCalledwith(pickedAns)
      expect(incrementQuestion).toHaveBeenCalledTimes(1)
      expect(incrementQuestion).toHaveBeenCalledWith(1)
      expect(setCurrentQuestion).toHaveBeenCalledTimes(1)
      expect(setCurrentQuestion).toHaveBeenCalledWith(questions[1])
      expect(getAnswers).toHaveBeenCalledTimes(1)
      expect(getAnswers).toHaveBeenCalledWith(body,"jwt")

    })
  })

  describe("addLastAnswer", () => {
    it("should call", () => {
      const lastQuestion = { id: 2, questionNumber: 2, description: "test2" }
      const quizProgressTracking = { questionNumber: 1, answersPicked:[], showResults: false }
      const component = render(
        <Provider store={store}>
          <Router history={history}>
            <QuizStart
              quizProgressTracking={quizProgressTracking}
              questions={questions}
              quiz={quiz}
              currentQuestion={lastQuestion}
              answers={answers}
              addAnswer={addAnswer}
              incrementQuestion={incrementQuestion}
              setCurrentQuestion={setCurrentQuestion}
              getAnswers={getAnswers}
              userData={userData}
              showResults={showResults}
              clearQuizProgress={clearQuizProgress}
              setNotification={setNotification}
            />
          </Router>
        </Provider>
      );
      console.log(questions.length)
      fireEvent.click(component.getByTestId("next-button"))
      expect(addAnswer).toHaveBeenCalledTimes(1)
      // // expect(addAnswer).toHaveBeenCalledwith()
      expect(incrementQuestion).toHaveBeenCalledTimes(1)
      // // expect(incrementQuestion).toHaveBeenCalledWith(2)
      expect(showResults).toHaveBeenCalledTimes(1)

    })
  })

})
