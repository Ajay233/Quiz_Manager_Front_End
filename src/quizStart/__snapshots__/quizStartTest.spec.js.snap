// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a question, list of answers to pick, 'next question' and 'quit' buttons 1`] = `
<div
  className="componentContainer"
>
  <div
    id="quizStartTitle"
  >
    test
  </div>
  <div
    id="quizStartContainer"
  >
    <div
      id="questionDetailsContainer"
    >
      <div
        className="sectionSpacing"
        id="questionContainer"
      >
        <div
          className="title-medium-left-alt"
        >
          Question 1
        </div>
        <div>
          test
        </div>
      </div>
      <img
        alt=""
        id="quizBot2"
        src="quizBot2.png"
      />
    </div>
    <div
      id="answerFormContainer"
    >
      <div
        id="pickAnswerForm"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            id="answerContainer"
          >
            <div
              className="title-medium-left-alt"
            >
              Answers
            </div>
            <div>
              <label>
                <input
                  checked={false}
                  className="radioInput"
                  name="answer"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  type="radio"
                  value="test"
                />
                test
              </label>
            </div>
            <div>
              <label>
                <input
                  checked={false}
                  className="radioInput"
                  name="answer"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  type="radio"
                  value="test2"
                />
                test2
              </label>
            </div>
          </div>
          <img
            alt=""
            src="reply.png"
          />
          <div
            className="buttonContainer"
          >
            <button
              className="submit quizButton-left"
              data-testid="next-button"
            >
              Next Question
            </button>
            <button
              className="submit quizButton-right"
              data-testid="quit-button"
              onClick={[Function]}
            >
              Quit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render the results 1`] = `
<div
  className="componentContainer"
>
  <div
    id="quizStartTitle"
  >
    test
  </div>
  <div
    id="quizStartContainer"
  >
    <div
      id="questionDetailsContainer"
    >
      <div
        className="sectionSpacing"
        id="questionContainer"
      >
        <div
          className="title-medium-left-alt"
        >
          Question 1
        </div>
        <div>
          test
        </div>
      </div>
      <img
        alt=""
        id="quizBot2"
        src="quizBot2.png"
      />
      <div
        className="title-large"
        id="quizResult"
      >
        <b>
          You scored 1 out of 2
        </b>
      </div>
    </div>
    <div
      id="answerFormContainer"
    >
      <div
        id="pickAnswerForm"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            id="answerContainer"
          >
            <div
              className="title-medium-left-alt"
            >
              Answers
            </div>
            <div>
              <label>
                <input
                  checked={false}
                  className="radioInput"
                  name="answer"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  type="radio"
                  value="test"
                />
                test
              </label>
            </div>
            <div>
              <label>
                <input
                  checked={false}
                  className="radioInput"
                  name="answer"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  type="radio"
                  value="test2"
                />
                test2
              </label>
            </div>
          </div>
          <img
            alt=""
            src="reply.png"
          />
          <div
            className="buttonContainer"
          >
            <button
              className="submit quizButton-left"
              data-testid="next-button"
            >
              Finish
            </button>
            <button
              className="submit quizButton-right"
              data-testid="quit-button"
              onClick={[Function]}
            >
              Quit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
